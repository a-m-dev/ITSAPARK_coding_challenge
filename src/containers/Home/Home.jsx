import "./Home.scss";

const Home = () => {
  return (
    <article className="container home">
      <section className="home__desc">
        <h1>ðŸ‘‹ Hello</h1>
        <h3> I provided three solution for this challenge!</h3>
        <p>
          You can chose each of them from navigation bar to check them how they
          work.
        </p>

        <ol>
          <li>
            First way is a simple modulo kind of checking number in an O(n)
            solution. It is not performant, and you might get your chrome tab
            crashed for large entries like 10 bilion.
          </li>
          <li>
            In second solution I am using worker to handle the computation and
            considering the large amount of input. I can say that it's a bit
            more performant.
          </li>
          <li>Third way is the mathmatical thinking way.</li>
        </ol>
      </section>
    </article>
  );
};

export default Home;
