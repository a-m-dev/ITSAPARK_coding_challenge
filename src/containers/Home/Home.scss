@import "../../globalStyles/variables.scss";
@import "../../globalStyles/mixins.scss";

.home {
  position: relative;
  height: calc(100vh - 200px);

  &__desc {
    position: relative;
    text-align: center;
    max-width: 458px;
    margin: 0 auto;
    padding: 0 $spacing-xxl;
    padding-top: $spacing-xxl * 3;
    @include makeFlex(center, center, column);

    & h1 {
      font-size: 3rem;
      font-weight: bold;
      color: $gray-800;
      margin-bottom: $spacing-sm;
    }

    & h3 {
      margin-bottom: $spacing-xl;
    }

    & p {
      text-align: left;
      margin-bottom: $spacing-xl;
    }

    & ol li {
      text-align: left;
      margin-bottom: $spacing-sm;
      color: $gray-500;
      line-height: 1.4;
    }
  }
}
