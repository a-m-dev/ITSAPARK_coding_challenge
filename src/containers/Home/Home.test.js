import { render } from "@testing-library/react";
import Home from "./Home";

describe("<Home />", () => {
  test("should match with snapshot", () => {
    const result = render(<Home />);
    expect(result).toMatchSnapshot();
  });
});
