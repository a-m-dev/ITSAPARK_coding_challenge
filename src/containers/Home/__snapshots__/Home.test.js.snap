// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should match with snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="container home"
      >
        <section
          class="home__desc"
        >
          <h1>
            Hello ðŸ‘‹ 
          </h1>
          <h3>
             I provided three solution for this challenge!
          </h3>
          <p>
            You can choose each of them from navigation bar to check them how they work.
          </p>
          <ol>
            <li>
              First way is a simple modulo kind of checking number in an O(n) solution. It is not performant as much, and you might get your chrome tab crashed for large entries like numbers more than 1 bilion.
            </li>
            <li>
              In second solution I am using a web worker to handle the large amount of computation in another thread. I can say that it's a bit more performant, and at least the UI is not locking and it becomes interactable.
            </li>
            <li>
              Third way is the algorithmic and mathmatical thinking way that I've implemented. which is the best solution that I could came up with!
            </li>
          </ol>
        </section>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="container home"
    >
      <section
        class="home__desc"
      >
        <h1>
          Hello ðŸ‘‹ 
        </h1>
        <h3>
           I provided three solution for this challenge!
        </h3>
        <p>
          You can choose each of them from navigation bar to check them how they work.
        </p>
        <ol>
          <li>
            First way is a simple modulo kind of checking number in an O(n) solution. It is not performant as much, and you might get your chrome tab crashed for large entries like numbers more than 1 bilion.
          </li>
          <li>
            In second solution I am using a web worker to handle the large amount of computation in another thread. I can say that it's a bit more performant, and at least the UI is not locking and it becomes interactable.
          </li>
          <li>
            Third way is the algorithmic and mathmatical thinking way that I've implemented. which is the best solution that I could came up with!
          </li>
        </ol>
      </section>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
